cmd/bosun/sched/host.go

```diff
@@ -0,0 +1,809 @@
+	if err != nil {
+		return nil, err
+	}
+	hwPowerSuppliesTags, err := s.Search.FilteredTagSets("hw.ps", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwTempsTags, err := s.Search.FilteredTagSets("hw.chassis.temps.reading", nil)
+	if err != nil {
+		return nil, err
+	}
+	hwBoardPowerTags, err := s.Search.FilteredTagSets("hw.chassis.power.reading", nil)
+	if err != nil {
+		return nil, err
+	}
+	diskTags, err := s.Search.FilteredTagSets("os.disk.fs.space_total", nil)
+	if err != nil {
+		return nil, err
+	}
+	// Will make the assumption that the metric bosun.ping.timeout, resolved, and rtt
+	// all share the same tagset
+	icmpTimeOutTags, err := s.Search.FilteredTagSets("bosun.ping.timeout", nil)
+	if err != nil {
+		return nil, err
+	}
+	for name, host := range hosts {
+		host.Name = name
+		hostTagSet := opentsdb.TagSet{"host": host.Name}
+		hostMetadata, err := s.GetMetadata("", hostTagSet)
+		if err != nil {
+			slog.Error(err)
+		}
+		processHostIncidents(host, states, silences)
+		for _, ts := range icmpTimeOutTags {
+			if ts["dst_host"] != host.Name {
+				continue
+			}
+			// The host tag represents the polling source for these set of metrics
+			source, ok := ts["host"]
+			if !ok {
+				slog.Errorf("couldn't find source tag for icmp data for host %s", host.Name)
+			}
+			// 1 Means it timed out
+			timeout, timestamp, err := s.Search.GetLast("bosun.ping.timeout", ts.String(), false)
+			if err != nil && !(timestamp > 0) {
+				continue
+			}
+			rtt, rttTimestamp, _ := s.Search.GetLast("bosun.ping.rtt", ts.String(), false)
+			// 1 means dns resolution was successful
+			dnsLookup, dnsTimestamp, dnsErr := s.Search.GetLast("bosun.ping.resolved", ts.String(), false)
+			host.ICMPData[source] = &ICMPData{
+				TimedOut:               timeout == 1 && err == nil,
+				TimedOutLastUpdated:    timestamp,
+				DNSResolved:            dnsLookup == 1 && dnsErr == nil,
+				DNSResolvedLastUpdated: dnsTimestamp,
+				RTTMS:          rtt,
+				RTTLastUpdated: rttTimestamp,
+			}
+
+		}
+		// Process Hardware Chassis States
+		for _, ts := range hwChassisTags {
+			if ts["host"] != host.Name {
+				continue
+			}
+			component, ok := ts["component"]
+			if !ok {
+				return nil, fmt.Errorf("couldn't find component tag for host %s", host.Name)
+			}
+			fstatus, timestamp, err := s.Search.GetLast("hw.chassis", ts.String(), false)
+			status := "Bad"
+			if fstatus == 0 {
+				status = "Ok"
+			}
+			if err == nil && timestamp > 0 {
+				host.Hardware.ChassisComponents[component] = &ChassisComponent{
+					Status:            status,
+					StatusLastUpdated: timestamp,
+				}
+			}
+		}
+		for _, ts := range hwTempsTags {
+			if ts["host"] != host.Name {
+				continue
+			}
+			name, ok := ts["name"]
+			if !ok {
+				slog.Errorf("couldn't find name tag %s for host %s", host.Name, name)
+			}
+			tStatus, timestamp, err := s.Search.GetLast("hw.chassis.temps", ts.String(), false)
```

Comment:
I notice you almost never check errors on these calls. Are you ok with zero values? Use _ if you aren't going to care about a particular error.